rotkehlchen/history/events/structures/asset_movement.py

```diff
@@ -0,0 +1,200 @@
+        `link`: For exchange asset movements this should be the exchange unique identifier.
+            For movements imported from third parties we should generate a unique id for this.
+            If movements are both imported from third parties like cointracking.info and from
+            the exchanges themselves then there is no way to avoid duplicates.
+        """  # noqa: E501
+        for key, value in {
+            'address': address,
+            'transaction_id': transaction_id,
+            'link': link,
+        }.items():
+            if value is None:
+                continue
+
+            if extra_data is None:
+                extra_data = {}
+
+            extra_data[key] = value
+
+        super().__init__(
+            event_identifier=event_identifier if event_identifier is not None else hash_id(
+                str(timestamp) +
+                str(location) +
+                str(event_type) +
+                str(event_subtype) +
+                asset.identifier +
+                str(link),
+            ),
+            sequence_index=sequence_index,
+            timestamp=timestamp,
+            location=location,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=asset,
+            balance=balance,
+            location_label=location_label,
+            notes=notes if notes is not None else self._create_event_notes(
+                location_name=get_formatted_location_name(location),
+                event_type=event_type,
+                event_subtype=event_subtype,
+                asset_symbol=asset.resolve_to_asset_with_symbol().symbol,
+                amount=balance.amount,
+            ),
+            identifier=identifier,
+            extra_data=extra_data,
+        )
+
+    @staticmethod
+    def _create_event_notes(
+            location_name: str,
+            event_type: HistoryEventType,
+            event_subtype: HistoryEventSubType,
+            asset_symbol: str,
+            amount: 'FVal',
+    ) -> str | None:
+        """Create event notes from the provided data.
+        Returns the generated notes or None on error."""
+        if event_type == HistoryEventType.DEPOSIT:
+            if event_subtype == HistoryEventSubType.DEPOSIT_ASSET:
+                return f'Deposit {amount} {asset_symbol} to {location_name}'
+            elif event_subtype == HistoryEventSubType.FEE:
+                return f'Deposit fee of {amount} {asset_symbol} on {location_name}'
+        elif event_type == HistoryEventType.WITHDRAWAL:
+            if event_subtype == HistoryEventSubType.REMOVE_ASSET:
+                return f'Withdraw {amount} {asset_symbol} from {location_name}'
+            elif event_subtype == HistoryEventSubType.FEE:
+                return f'Withdrawal fee of {amount} {asset_symbol} on {location_name}'
+
+        log.error('Unknown asset movement type/subtype encountered, leaving event notes blank.')
```

Comment:
But this should actually never happen. Since we control how those are initialized. Right?